topology/probes/libvirt/libvirt.go

```diff
@@ -106,45 +131,122 @@ func (probe *Probe) getDomainInterfaces(
 		return
 	}
 	for _, itf := range d.Interfaces {
-		if constraint == "" || constraint == itf.Alias.Name {
+		if constraint == "" || (itf.Alias != nil && constraint == itf.Alias.Name) {
 			itf.Host = domainNode
 			itfObj := itf
 			interfaces = append(interfaces, &itfObj)
 		}
 	}
+	for _, hostdev := range d.HostDevices {
+		logging.GetLogger().Debugf("Found Hostdev %v", hostdev)
+		if hostdev.Mode != "subsystem" || hostdev.Type != "pci" {
+			continue
+		}
+		if hostdev.Source == nil || hostdev.Source.Address == nil {
+			continue
+		}
+		hostdev.Host = domainNode
+		hostdevs = append(hostdevs, &hostdev)
+	}
+	return
+}
+
+func (probe *Probe) makeHostDev(name string, source *Source) (node *graph.Node, err error) {
+	if source == nil || source.Address == nil {
+		err = errors.New("hostdev without PCI address")
+		return
+	}
+	address := formatPciAddress(source.Address)
+	node = probe.graph.LookupFirstChild(probe.root, graph.Metadata{
+		"BusInfo": address,
+	})
+	if node == nil {
+		m := graph.Metadata{
+			"Name":    fmt.Sprintf("%s-%s", name, strings.TrimPrefix(address, "0000:")),
+			"BusInfo": address,
+			"Driver":  "pci-passthrough",
```

Comment:
You added the driver field in the XML parser, could we use it here ?

Reply:
no: the driver field is used when the libvirt presents an abstraction of the device to the VM. Here it just gives a pci descriptor after eventual translation of the BDF (that is why you still have a pci address that is often different from the source address).  In my case the driver would be i40evf but libvirt does not even know this.